include/circt/Dialect/RTL/RTLCombinatorial.td

```diff
@@ -339,3 +339,53 @@ def MuxOp : RTLOp<"mux",
     }]>
   ];
 }
+
+def BitArrayType : Type<CPred<[{
+  $_self.isa<ArrayType>() &&
+  $_self.cast<ArrayType>().getElementType().isSignlessInteger(1)
+}]>, "RTL array of bits">;
+
+def KnownBitWidthType : Type<CPred<[{getBitWidth($_self) != -1}]>,
+  "Type wherein the bitwidth in hardware is known">;
+
+def CastToBitsOp: RTLOp<"cast.tobits", [NoSideEffect]> {
+  let summary = "Cast an arbitrary value to a bit array";
```

Comment:
I'd recommend:

Cast one value to to another value of the same size and potentially different type, using a bitwise reinterpretation of the value.